src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -961,31 +966,68 @@ void FancyZones::HandleVirtualDesktopUpdates(HANDLE fancyZonesDestroyedEvent) no
         {
             return;
         }
-        const int guidSize = sizeof(GUID);
-        std::unordered_map<GUID, bool> temp;
-        temp.reserve(bufferCapacity / guidSize);
-        for (size_t i = 0; i < bufferCapacity; i += guidSize)
-        {
-            GUID* guid = reinterpret_cast<GUID*>(buffer.get() + i);
-            temp[*guid] = true;
-        }
-        std::unique_lock writeLock(m_lock);
-        for (auto it = begin(m_virtualDesktopIds); it != end(m_virtualDesktopIds);)
+        RegisterVirtualDesktopUpdates(buffer, bufferCapacity);
+    }
+}
+
+void FancyZones::RegisterVirtualDesktopUpdates(std::unique_ptr<BYTE[]>& buffer, int bufferCapacity) noexcept
+{
+    const int guidSize = sizeof(GUID);
+    std::unordered_set<GUID> temp;
+    temp.reserve(bufferCapacity / guidSize);
+    for (size_t i = 0; i < bufferCapacity; i += guidSize)
+    {
+        GUID* guid = reinterpret_cast<GUID*>(buffer.get() + i);
+        temp.insert(*guid);
+    }
+    std::unique_lock writeLock(m_lock);
+    for (auto it = begin(m_processedWorkAreas); it != end(m_processedWorkAreas);)
+    {
+        auto iter = temp.find(it->first);
+        if (iter == temp.end())

```

Comment:
let's add a comment with something like "if we couldn't find the GUID in currentVirtualDesktopIds, we must remove it from both m_processedWorkAreas and deviceInfoMap"